src/common/downloadFileDialog.component.tsx

```diff
@@ -0,0 +1,103 @@
+import WarningIcon from '@mui/icons-material/Warning';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormHelperText,
+} from '@mui/material';
+import { UseQueryResult } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import React from 'react';
+import { APIImage, APIImageWithURL } from '../api/api.types';
+
+export interface BaseDownloadFileProps {
+  open: boolean;
+  onClose: () => void;
+  fileType: 'Image' | 'Attachment';
+}
+
+export interface ImageDownloadDialogProps extends BaseDownloadFileProps {
+  fileType: 'Image';
+  useGetFile: (id: string) => UseQueryResult<APIImageWithURL, AxiosError>;
+  file: APIImage;
+}
+
+export type DownloadFileProps = ImageDownloadDialogProps;
+
+const DownloadFileDialog = (props: DownloadFileProps) => {
+  const { open, onClose, fileType, file, useGetFile } = props;
+
+  const [formError, setFormError] = React.useState<string | undefined>(
+    undefined
+  );
+
+  const { data: downloadedFile, isLoading: downloadIsLoading } = useGetFile(
+    file.id
+  );
+
+  const handleClose = React.useCallback(() => {
+    setFormError(undefined);
+    onClose();
+  }, [onClose]);
+
+  const handleDownloadFile = React.useCallback(() => {
+    if (downloadedFile) {
+      const link = document.createElement('a');
+      link.href = downloadedFile.download_url;
+      link.download = downloadedFile.file_name;
+      document.body.appendChild(link);
```

Comment:
Can you change this into a general component (downloadFile funciton) in the utils.tsx